### PR TITLE
Feature/12 login page android

### PR DESCRIPTION
**Pull request by sebastian-vangrieken**

### CODE REVIEW

File: composeApp/build.gradle.kts [hunk 1]
```diff
@@ -82,4 +82,7 @@ android {
     dependencies {
         debugImplementation(libs.compose.ui.tooling)
     }
-}
\ No newline at end of file
+}
+dependencies {
+    implementation(libs.androidx.media3.common)
```
Comment (DanneD-Wisemen) [inline composeApp/build.gradle.kts]: Why are these in a new dependencies block and not in the above block?

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/MainActivity.kt [hunk 2]
```diff
@@ -10,30 +9,17 @@ import androidx.compose.foundation.layout.fillMaxSize
...
-import db.repository.reservation.RemoteReservationRepository
-import db.repository.user.RemoteUserRepository
 import org.koin.core.component.KoinComponent
-import org.koin.core.component.inject
 
 class MainActivity : ComponentActivity(), KoinComponent {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        val userRepo: RemoteUserRepository by inject()
```
Comment (DanneD-Wisemen) [inline composeApp/src/androidMain/kotlin/com/wisemen/chargehub/MainActivity.kt:32]: Why are all these deleted?

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/components/TextFields.kt [hunk 3]
```diff
@@ -143,7 +155,7 @@ object TextFields {
 
         Box(modifier, contentAlignment = Alignment.CenterEnd) {
             EditText(
-                errorMessage = if (!isValid) stringResource(R.string.password_is_required) else null,
+                errorMessage = if (! isValid) stringResource(R.string.password_is_required) else null,
```
Comment (DanneD-Wisemen) [inline composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/components/TextFields.kt]: Why are these spaces added?

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/components/TextFields.kt [hunk 4]
```diff
@@ -182,7 +194,7 @@ object TextFields {
     ) {
         EditText(
             modifier = modifier,
-            errorMessage = if (!isValid) stringResource(R.string.email_is_required) else null,
+            errorMessage = if (! isValid) stringResource(R.string.email_is_required) else null,
```
Comment (DanneD-Wisemen) [inline composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/components/TextFields.kt]: Why are these spaces added?

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/LoginScreen.kt [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+        },
+    ) { padding ->
+        Column(
+            modifier = Modifier.padding(padding).fillMaxWidth(),
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+            Text(
+                stringResource(R.string.welkom_terug),
+                style = TextStyles.title,
+                modifier = Modifier.padding(bottom = Spacing.small, top = 70.dp)
```
Comment (DanneD-Wisemen) [inline composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/LoginScreen.kt]: Try and keep modifiers at the top, check this for all your components

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/LoginScreen.kt [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+                    isValid = true,
+                )
+                Text(
+                    text = stringResource(R.string.wachtwoord_vergeten),
+                    style = TextStyles.small_text,
+                    modifier = Modifier
+                        .padding(bottom = 30.dp, start = 15.dp)
+                        .clickable {
+                            onAction(LoginScreenUiAction.OnClickedForgotPasswordButtonAction)
+                            Log.d("Email", state.email)
```
Comment (DanneD-Wisemen) [inline composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/LoginScreen.kt]: Logs that should be removed

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/LoginScreen.kt [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+                                imageVector = Icons.Filled.ChevronLeft,
+                                contentDescription = null
+                            )
+                        }
+                    }
+                },
+                colors = TopAppBarDefaults.topAppBarColors(containerColor = Colors.white),
+            )
+        },
+    ) { padding ->
```
Comment (DanneD-Wisemen) [inline composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/LoginScreen.kt]: ContentPadding would be a better name

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/LoginScreen.kt [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+                stringResource(R.string.aanmelden_welkom_tekst),
+                style = TextStyles.text,
+                modifier = Modifier.padding(bottom = 30.dp)
+            )
+            Column {
+                TextFields.EmailTextField(
+                    email = state.email,
+                    isValid = true,
+                    onInputChanged = { onAction(LoginScreenUiAction.OnEmailChangedAction(it)) },
+                    modifier = Modifier.padding(horizontal = 15.dp).padding(bottom = 5.dp)
```
Comment (DanneD-Wisemen) [inline composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/LoginScreen.kt]: This is a mistake of the designer, these horizontalpaddings usually are 16dp

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/LoginScreen.kt [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+                )
+                TextFields.PasswordTextField(
+                    modifier = Modifier.padding(horizontal = 15.dp),
```
Comment (DanneD-Wisemen) [inline composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/LoginScreen.kt]: The same here for the 16dp

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/LoginScreen.kt [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+                    },
+                    currentPassword = state.password,
+                    isPasswordVisible = state.passwordVisibility,
+                    isValid = true,
+                )
+                Text(
+                    text = stringResource(R.string.wachtwoord_vergeten),
+                    style = TextStyles.small_text,
+                    modifier = Modifier
+                        .padding(bottom = 30.dp, start = 15.dp)
```
Comment (DanneD-Wisemen) [inline composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/LoginScreen.kt]: 16dp

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/LoginScreen.kt [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+                            Log.d("Email", state.email)
+                            Log.d("Password", state.password)
+                        }
+                )
+            }
+            Buttons.PrimaryButton(
+                text = stringResource(R.string.aanmelden),
+                colors = ButtonDefaults.primaryButtonColors(),
+                onClick = { onAction(LoginScreenUiAction.OnClickedLoginButtonAction) },
+                modifier = Modifier.fillMaxWidth().padding(horizontal = 15.dp)
```
Comment (DanneD-Wisemen) [inline composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/LoginScreen.kt]: 16dp

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/LoginScreen.kt [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+                    modifier = Modifier
+                        .padding(bottom = 30.dp, start = 15.dp)
+                        .clickable {
+                            onAction(LoginScreenUiAction.OnClickedForgotPasswordButtonAction)
+                            Log.d("Email", state.email)
+                            Log.d("Password", state.password)
+                        }
+                )
+            }
+            Buttons.PrimaryButton(
```
Comment (DanneD-Wisemen) [inline composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/LoginScreen.kt:157]: add a white line between components

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/LoginScreen.kt [hunk 13]
```diff
@@ -0,0 +1,163 @@
...
+        Column(
+            modifier = Modifier.padding(padding).fillMaxWidth(),
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+            Text(
+                stringResource(R.string.welkom_terug),
+                style = TextStyles.title,
+                modifier = Modifier.padding(bottom = Spacing.small, top = 70.dp)
+            )
+            Text(
```
Comment (DanneD-Wisemen) [inline composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/LoginScreen.kt:122]: add a white line between components

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/theme/Spacing.kt [hunk 14]
```diff
@@ -0,0 +1,8 @@
+package com.wisemen.chargehub.ui.theme
+
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+
+object Spacing {
+    val small: Dp = 8.dp
```
Comment (DanneD-Wisemen) [inline composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/theme/Spacing.kt:7]: add a spacing medium: 16.dp

File: composeApp/src/main/res/values/strings.xml [hunk 15]
```diff
@@ -9,6 +9,9 @@
     <string name="charge">Charge\nHub.</string>
     <string name="aanmelden">Aanmelden</string>
     <string name="registreren">Registreren</string>
+    <string name="welkom_terug">Welkom Terug!</string>
```
Comment (DanneD-Wisemen) [inline composeApp/src/main/res/values/strings.xml]: try and keep these keys in English please

File: shared/build.gradle.kts [hunk 16]
```diff
@@ -3,6 +3,7 @@ plugins {
     alias(libs.plugins.androidLibrary)
     alias(libs.plugins.kotlinSerialization)
     alias(libs.plugins.sqlDelight)
+    id("com.rickclephas.kmp.nativecoroutines") version "1.0.0-ALPHA-26"
```
Comment (DanneD-Wisemen) [inline shared/build.gradle.kts:6]: Use the version catalog

**Review by DanneD-Wisemen — APPROVED** [review]

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/MainActivity.kt (hunk 2 above)
Comment (sebastian-vangrieken) [reply composeApp/src/androidMain/kotlin/com/wisemen/chargehub/MainActivity.kt:32]: They weren't used.

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/components/TextFields.kt (hunk 3 above)
Comment (sebastian-vangrieken) [reply composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/components/TextFields.kt]: Probably the android studio code clean-up

Comment (DanneD-Wisemen) [reply composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/components/TextFields.kt]: Weird, check if you can fix this. This is weird formatting

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/LandingScreen.kt [hunk 17]
```diff
@@ -95,7 +94,7 @@ fun LandingLayout(onAction: (LandingScreenUiAction) -> Unit) {
 fun Header() {
     Column {
         Text(
-            stringResource(R.string.stroomlijn_je_dag_met),
+            stringResource(R.string.stroomline_your_day_with),
```
Comment (DanneD-Wisemen) [inline composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/LandingScreen.kt]: Streamline 😄

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/LoginScreen.kt (hunk 5 above)
Comment (DanneD-Wisemen) [reply composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/LoginScreen.kt]: Keep this in mind

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/LoginScreen.kt (hunk 6 above)
Comment (DanneD-Wisemen) [reply composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/LoginScreen.kt]: Don't forget these

File: composeApp/src/main/res/values/strings.xml [hunk 18]
```diff
@@ -5,16 +5,16 @@
     <string name="email_is_required">" 'Email' is invalid"</string>
     <string name="email">Email</string>
     <string name="password">Password</string>
-    <string name="stroomlijn_je_dag_met">Stroomlijn je dag met</string>
+    <string name="stroomline_your_day_with">Stroomlijn je dag met</string>
     <string name="charge">Charge\nHub.</string>
-    <string name="aanmelden">Aanmelden</string>
+    <string name="login">Aanmelden</string>
     <string name="registreren">Registreren</string>
```
Comment (DanneD-Wisemen) [inline composeApp/src/main/res/values/strings.xml]: Register

File: composeApp/src/main/res/values/strings.xml [hunk 19]
```diff
@@ -5,16 +5,16 @@
...
-    <string name="welkom_terug">Welkom Terug!</string>
-    <string name="aanmelden_welkom_tekst">Fijn om je terug te zien</string>
-    <string name="wachtwoord_vergeten">Wachtwoord vergeten?</string>
+    <string name="welcome_back">Welkom Terug!</string>
+    <string name="login_welcome_text">Fijn om je terug te zien</string>
+    <string name="forgot_password">Wachtwoord vergeten?</string>
     <string name="privacybeleid">Privacybeleid</string>
```
Comment (DanneD-Wisemen) [inline composeApp/src/main/res/values/strings.xml]: privacy_policy

**Review by DanneD-Wisemen — CHANGES_REQUESTED** [review]

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/components/TextFields.kt (hunk 3 above)
Comment (sebastian-vangrieken) [reply composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/components/TextFields.kt]: I corrected it

**Pull request merged**